### ISSUE TITLE
soap11.ValueError: lxml decode utf-8 error

### ISSUE THREAD

**Issue by cpwnd** [arskom/spyne#503]
Refering to the stackoverflow question http://stackoverflow.com/questions/38379521

My spyne instance (2.12.11) throwing ValueError's, when getting requests. So here is my test case with curl and a test server.

Server code:

    #!/usr/bin/python
    # encoding: utf8
    from spyne import InternalError
    from spyne import error
    
    from spyne.model.complex import Unicode
    from spyne.model.complex import Iterable
    from spyne.service import ServiceBase
    from spyne.decorator import rpc
    from spyne.application import Application
    from spyne.protocol.soap import Soap11
    from spyne.server.wsgi import WsgiApplication
    
    import logging
    logging.basicConfig(level=logging.DEBUG)
    
    class SimpleService(ServiceBase):
    
        @rpc(Unicode, Unicode, _returns=Iterable(Unicode))
        def call(ctx, arg1, arg2):
            yield arg1
            yield arg2
    
    application = Application([SimpleService],
                             name='SimpleService',
                             tns='badass_ns',
                             in_protocol=Soap11(validator='lxml'),
                             out_protocol=Soap11()
                             )

    if __name__ == '__main__':
        from wsgiref.simple_server import make_server

        wsgi_app = WsgiApplication(application)
        server = make_server('0.0.0.0', 8000, wsgi_app)
        server.serve_forever()
        
With this as a xml request code:

    <?xml version="1.0" encoding="utf-8"?>
    <soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"
                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                   xmlns:xsd="http://www.w3.org/2001/XMLSchema">
        <soap:Body>
            <call xmlns="badass_ns">
                <arg1>Hello</arg1>
                <arg2>World</arg2>
            </call>
        </soap:Body>
    </soap:Envelope>

Firing up with this terminal command:

    curl -X POST -H "Content-Type: text/xml; charset=utf-8" -d @request.xml localhost:8000

Console output:

    DEBUG:spyne.protocol.soap.soap11:ValueError: Deserializing from unicode strings
    with encoding declaration is not supported by lxml.
    DEBUG:spyne.protocol.xml:Validated ? False
    40.50.60.70 - - [14/Jul/2016 16:58:42] "POST /?WSDL HTTP/1.1" 500 480

If this helps, I can say, without the `charset=utf-8` in the POST header, the valueerror won't appear.

**Comment by rich5851** [comment]
Same issue for me. If the POST header declares charset=utf-8 and then the xml request also has encoding="utf-8" set then I get the error. Removing it solves the problem however I have no control over the client that is sending in the requests.

**Comment by rich5851** [comment]
I figured it out was due to using the simple server to run the wsgi. I changed it to use twisted to serve the application and the problem was solved.   I think the problem is with the server dumping Unicode straight into lxml for the parsing. Twisted seems to handle this better and gives libxml something more palatable to deal with. Having the encoding declared in both post header and request is no longer a problem.
